vault/diagnose/storage_checks.go

```diff
@@ -80,3 +81,14 @@ func StorageEndToEndLatencyCheck(ctx context.Context, b physical.Backend) error
 	}
 	return nil
 }
+
+// ConsulDirectAccess verifies that consul is connecting to local agent,
+// versus directly to a remote server. We can only assume that the local address
+// is a server, not a client.
+func ConsulDirectAccess(config map[string]string) string {
+	serviceRegistrationAddr := config["address"]
```

Comment:
do we want to handle the case where `address` isn't in `config`?

Reply:
Yes, good point!